### ISSUE TITLE
Move `org.typelevel` to `s01.oss.sonatype.org`

### ISSUE THREAD

**Issue by armanbilge** [typelevel/general#114]
Wanted to document more permanently @bpholt's excellent suggestion. This is currently the recommendation at https://status.maven.org/:

<img width="874" alt="Screen Shot 2021-12-16 at 13 00 30" src="https://user-images.githubusercontent.com/3119428/146448372-8605e757-7aeb-4f87-b58b-bd309cbefce9.png">

> Investigating - Staging operations continue to run slow, and the current load on oss.sonatype.org isn't decreasing any time soon (which would allow the backlog to clear up). Please consider migrating to our new, much less loaded host. See here for context: https://twitter.com/sonatype_ops/status/1471099908300189705

Note that this will require changing the _publish_ config for all typelevel projects to the new host (the change is completely transparent to dependencies using the artifacts i.e. no need to add additional resolvers).
```scala
sonatypeCredentialHost := "s01.oss.sonatype.org"
```

@tpolecat documented how to make this change [on Twitter](https://twitter.com/tpolecat/status/1471580886827249670).

Example build with this config (note _all_ the places it is set):
https://github.com/armanbilge/schrodinger/blob/0897f5a07c5f02fd00b20c171590812906e16984/build.sbt

**Comment by armanbilge** [comment]
I didn't even scroll down on the status page the first time ... apparently s01 is a whopping 25x faster (!).

![Screen Shot 2021-12-16 at 5 30 46 PM](https://user-images.githubusercontent.com/3119428/146458706-75d1ec6e-2645-414d-ae06-eba0eee7d180.png)

**Comment by joroKr21** [comment]
`sbt-github-actions` has this note: https://github.com/xerial/sbt-sonatype#buildsbt
Maybe it's time to change the default? But idk why it matters when the account was created 🤔

**Comment by armanbilge** [comment]
> But idk why it matters when the account was created 🤔

Sonatype has been putting all new registrations at s01 since February 2021. That's why I'm on s01.

**Comment by joroKr21** [comment]
> Sonatype has been putting all new registrations at s01 since February 2021. That's why I'm on s01.

Does it mean that old accounts can't use `s01`?

**Comment by armanbilge** [comment]
No, it seems there is an open invitation for everyone to move to s01. From twitter:
> Central Publishers, extreme load and timeouts are reminiscent of the Bintray shutdown earlier this year, when we invited publishers to migrate to our new host.  That invite is very much still open.  Please create a ticket if you're interested

https://twitter.com/sonatype_ops/status/1471099908300189705

IIUC Rob requested to be moved.

**Comment by joroKr21** [comment]
Ah ok got it, so it requires a migration of the account - then it can't be the default

**Comment by armanbilge** [comment]
Aha, it seems I was mistaken, there _is_ some flexibility about when we lose access to the legacy host.

> We normally coordinate with the publisher before turning off access on the legacy host, though we strongly recommend that publishers make a clean break as soon as possible.

https://twitter.com/sonatype_ops/status/1471658405651795971

**Comment by armanbilge** [comment]
Another update: @djspiewak mentioned that `org.typelevel` may already be setup for `s01`. Seems it is, and I was just able to publish https://github.com/typelevel/feral to there. So I guess projects can move at their leisure.

**Comment by armanbilge** [comment]
Umm ... I guess they _don't_ support both hosts, and they just revoked our access to `s01`. Woops.
https://issues.sonatype.org/browse/OSSRH-76765

> I think that we had a mismatch here by our automated system.

> Your groupId was already registered in oss.sonatype.org before (OSSRH-4953) but you asked for a migration to the new host s01 (OSSRH-67693), but it was incomplete.

> Some of yours users had permissions on s01 but I just removed them to avoid any future issues (it's not supported to publish on both hosts at the same time). So I rolled back the incomplete migration.

> Do you want to be migrated to the new host s01.oss.sonatype.org? (Access to oss.sonatype.org for this groupId will be removed, changes at your side in your pom.xml files needs to be made and if you are using user tokens those will be need to be recreated in the new host)

> Also I rolled back the incomplete migration for co.fs2, for that groupId please create a new independent ticket if you want to migrate to the new host s01.oss.sonatype.org

So, like, it would be still good to do this 😅

**Comment by rossabaker** [comment]
I still don't understand what action is required as myself, as a Typelevel build maintainer, and as a non-Typelevel build maintainer.  When they migrate Typelevel, and Typelevel publishers, are we still going to be able to publish non-Typelevel projects?

**Comment by armanbilge** [comment]
To the best of my understanding:
1. Asking them to switch Typelevel to s01.
2. Updating the settings in all our builds to point to s01.
3. If you are using generated sonatype credentials, generating a new set at s01 and updating those secrets.

I was surprised about (3). I am juggling two sets of sonatype credentials, one pair for publishing Typelevel to the legacy host and another pair for publishing to my personal groupid at s01.

> are we still going to be able to publish non-Typelevel projects

Yes. I think they migrate you specifically for that groupid. Which is why I have credentials for both hosts.

**Comment by rossabaker** [comment]
Oof.  I set up automated publishing for several projects, so a bunch of them are going to start failing on publish.  I guess we'll address them as they fail, but this could be a long tail of aggravation.  Seems like staying put will also be increasingly aggravating.

Yeah, :+1:, whenever.

**Comment by armanbilge** [comment]
I'm hoping the answer is a combination of moving as many projects as possible to sbt-typelevel, and maybe setting some organization-level secret Sonatype credentials.

Then, we can ask sonatype to migrate us, swap out the credentials once, cut a new release of sbt-typelevel with the default changed, and we should be on our merry way.

**Comment by rossabaker** [comment]
That sounds good to me.  What I'd really like is to have multiple Sonatype API keys, so we can revoke one without redoing all the builds, but that wasn't possible last time I looked.

**Comment by armanbilge** [comment]
> so we can revoke one without redoing all the builds

This is exactly what we could do with organization-level secrets: change them for all the builds at once.

**Comment by rossabaker** [comment]
Oh, I see.  GitHub-org level.  I was thinking of a dummy Sonatype account, which is still a good idea: I don't want to use one that can publish to any other orgs.  We can't make the blast radius smaller than Typelevel, but we can make it no bigger.

**Comment by armanbilge** [comment]
Yep, my thinking too. Fs2 seems to have one:
https://issues.sonatype.org/secure/ViewProfile.jspa?name=fs2-ci

**Comment by rossabaker** [comment]
Winding down this repo.  Transferring to sbt-typelevel.
